scripts/irods/test/test_izonereport.py

```diff
@@ -86,14 +94,78 @@ def test_izonereport_and_validate(self):
     def test_resource_json_has_id(self):
         with session.make_session_for_existing_admin() as admin:
             _, stdout, _ = admin.assert_icommand(['izonereport'], 'STDOUT')
-            icat_server_object = json.loads(stdout)['zones'][0]['icat_server']
-
-            self.assertIn('resources', icat_server_object.keys())
-            self.assertGreaterEqual(len(icat_server_object['resources']), 1)
-            for resource in icat_server_object['resources']:
+            
+            zone_info = json.loads(stdout)['zones'][0]
+            server_array = zone_info['servers']
+            catalog_server = server_array[0]
+            for server in server_array:
+                if server['server_config']['catalog_service_role'] == 'provider':
+                    catalog_server = server
+                    break
+                
+            self.assertIn('resources', catalog_server.keys())
+            self.assertGreaterEqual(len(catalog_server['resources']), 1)
+            for resource in catalog_server['resources']:
                 self.assertIn('id', resource.keys())
 
-            self.assertIn('coordinating_resources', icat_server_object.keys())
-            self.assertGreaterEqual(len(icat_server_object['coordinating_resources']), 1)
-            for resource in icat_server_object['coordinating_resources']:
+            self.assertIn('coordinating_resources', zone_info.keys())
+            self.assertGreaterEqual(len(zone_info['coordinating_resources']), 1)
+            for resource in zone_info['coordinating_resources']:
                 self.assertIn('id', resource.keys())
+
+    # see issue #6857
+    @unittest.skipIf(plugin_name == 'irods_rule_engine_plugin-python' or test.settings.RUN_IN_TOPOLOGY, "Skip for Topology Testing or python rule engine")
+    def test_servers_are_flattened_core(self):
```

Comment:
As a convention, we include the issue number in the test's name like so.
```python
def test_servers_are_flattened_core__issue_6857(self):
```

Then you can remove the comment.